module-4/backend/week_one.md

```diff
@@ -1,21 +1,36 @@
 ## Week One - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's the most useful thing you learned from completing the intermission week work?
++ Understanding that way I would approach problems in Ruby can be applied to JavaScript.
 2. What are some tools to help debug JavaScript code?
++ eval(pry.it), debugger and console.log
 3. What are some tools you need in order to unit test your JavaScript?
++ Chai and Mocha
 4. What is the syntax for invoking a function?
++ functionName();
 5. What's the difference between `==` and `===` in JavaScript?
-6. What's the difference between asynchronous and synchronous JavaScript? 
++ `==` uses type conversion when comparing and `===` does not.  
+6. What's the difference between asynchronous and synchronous JavaScript?
++ Asynchronous means that it can happen while other functions are performed, while synchronous means that it will
+perform inline with everything else.
 7. What's a callback function and what are some reasons when we use/need callback functions?
++ Callbacks are anonymous functions within functions `$(this).on('click', function(){})`. They are useful because they follow logical design patterns like 'this then that'.   
 8. What's the biggest difference between a promise and a callback?
++ A promise object is what many asynchronous functions return (like an AJAX request) and we use callbacks on these objects to further manipulate elements.
 9. How do we setup a route when creating an API with Node and Express?
 10. What's `npm` and what do we use it for?
++ It is the default package manager for node.js and allows developers to download useful add-ons for their application.  
 
 #### Review  
 11. What's the MVC design pattern? Describe each part of MVC?
++ Model - Contains all the information a particular model needs to operate in your program.
```

Comment:
i encourage you to elaborate on this answer and don't use the different pieces while explaining each. what do they do individually and how do they relate to each other? knowing how to describe the MVC pattern is important 👍